include/logging/log_core.h

```diff
@@ -15,6 +15,10 @@
 extern "C" {
 #endif
 
+#ifdef __LP64__
```

Comment:
In #9522 the pointers are 32-bit even though the system is running in 64 bit. Would this check work with that PR?

Reply:
CC @andyross @andrewboie

Reply:
__LP64__ by definition, indicates that longs and pointers are 64-bit.
http://www.unix.org/version2/whatsnew/lp64_wp.html

I haven't tested this, but since the x86_64 port runs with 32-bit pointers, I imagine we should be fine, the compiler shouldn't be defining this macro.

Reply:
Yeah, x86_64 won't hit this because it's running in the x32 ABI.  But... I mean, it's x32 until it's not.  A very reasonable future would have us running in a fully 64 bit mode on platforms like x86 and RISC-V.  Not a reason to reject this patch if there are known word size issues, but we'll eventually want to fix the underlying issue.